### PR TITLE
 CE-2769: Add TemplateClassificationStorage client

### PR DESCRIPTION
**Pull request by lukaszkonieczny**
Classes in `Swagger` directory are auto generated by `swagger-codegen` so shouldn't be code reviewed. These code also shouldn't be modified.

### CODE REVIEW

File: includes/wikia/api/TemplateClassificationApiController.class.php [hunk 1]
```diff
@@ -0,0 +1,177 @@
+<?php
+
+use Wikia\Service\Gateway\ConsulUrlProvider;
+use Wikia\Service\Swagger\ApiProvider;
+use Swagger\Client\TemplateClassification\Storage\Api\TCSApi;
+use Swagger\Client\TemplateClassification\Storage\Models\TemplateTypeProvider;
+use Swagger\Client\ApiException;
+
+class TemplateClassificationApiController extends WikiaApiController
+{
```
Comment (Grunny) [inline includes/wikia/api/TemplateClassificationApiController.class.php]: Curly brace should be on the same line as the class definition per conventions.

File: includes/wikia/api/TemplateClassificationApiController.class.php [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+	 *
+	 * @param int $wikiId
+	 * @param int $pageId
+	 * @return string
+	 * @throws ApiException
+	 * @throws BadRequestApiException
+	 * @throws Exception
+	 */
+	public function getType( $wikiId, $pageId ) {
+		$templateType = "";
```
Comment (Grunny) [inline includes/wikia/api/TemplateClassificationApiController.class.php]: Should be single quotes per conventions.

File: includes/wikia/api/TemplateClassificationApiController.class.php [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+	 * @throws BadRequestApiException
+	 * @throws Exception
+	 */
+	public function getDetails( $wikiId, $pageId ) {
+		$templateDetails = [];
+
+		try {
+			$details = $this->getApiClient()->getTemplateDetails( $wikiId, $pageId );
+
+			if ( !is_null($details) ) {
```
Comment (Grunny) [inline includes/wikia/api/TemplateClassificationApiController.class.php]: Should be spacey per conventions.

File: includes/wikia/api/TemplateClassificationApiController.class.php [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+	 * @return TCSApi
+	 */
+	private function createApiClient() {
+		global $wgConsulUrl, $wgConsulServiceTag;
+		$urlProvider = new ConsulUrlProvider( $wgConsulUrl, $wgConsulServiceTag );
+		$apiProvider = new ApiProvider( $urlProvider );
+		return $apiProvider->getApi( self::SERVICE_NAME, TCSApi::class );
+	}
+
+}
```
Comment (Grunny) [inline includes/wikia/api/TemplateClassificationApiController.class.php]: Needs a newline at the end of file.

**Comment by Grunny** [comment]
Shouldn't TemplateClassificationApiController.class.php be defined in the setup file so it's accessible?

File: includes/wikia/api/TemplateClassificationApiController.class.php [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+	private function validateRequest( User $user, WikiaRequest $request ) {
+		if ( !$request->wasPosted() ) {
+			throw new BadRequestApiException();
+		}
+
+		if ( !$user->isLoggedIn() || !$user->matchEditToken( $request->getVal( 'editToken' ) ) ) {
+			throw new UnauthorizedException();
+		}
+
+		if ( !$user->isAllowed( 'edit' ) ) {
```
Comment (Grunny) [inline includes/wikia/api/TemplateClassificationApiController.class.php]: Should we be checking if they can edit the page they are classifying? Which would also check if the user is blocked.

**Comment by Grunny** [comment]
LGTM, other than the last comment about including the class in the setup file.

**Comment by lukaszkonieczny** [comment]
I assume that we include this file when we need to use it in the code.

**Comment by Grunny** [comment]
Fair enough then. :)

**Pull request merged**